plugins/inputs/snmp/snmp.go

```diff
@@ -415,6 +415,9 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 			// index, and being added on the same row.
 			if pkt, err := gs.Get([]string{oid}); err != nil {
 				return nil, fmt.Errorf("performing get on field %s: %w", f.Name, err)
+			} else if pkt != nil && pkt.PDUType == gosnmp.Report {
+				_, _, oidText, _, _ := SnmpTranslate(pkt.Variables[0].Name)
+				return nil, fmt.Errorf("recieved %s report", strings.TrimSuffix(oidText, ".0"))
```

Comment:
I'm new to SNMP v3 report PDUs.  As I understand it there are only a few reasons telegraf would receive a report PDU and one of the reasons is an auth error.  An auth error is very likely due to the user misconfiguring a password.

I think the error message here should suggest the user check their SNMP auth config.  I like how #8915 did this with "received SNMP report instead of response for field %s: check authentication?"

Reply:
The only reason for my PR instead of the original one was to be able to give a more specific message to the user. Just saying a report received will not help much for a regular user. This is also the reason I try to translate the report OID so I could mention what exactly the report was about. 

In my opinion, If a user sees one of these names in the log message, they might understand what the problem is.
- usmStatsUnsupportedSecLevels
- usmStatsNotInTimeWindows
- usmStatsUnknownUserNames
- usmStatsUnknownEngineIDs
- usmStatsWrongDigests
- usmStatsDecryptionErrors

(some of them, the user will never see since gosnmp is already handling them)

And these are only the ones that occur when having a bad auth setup, I think there are more possibilities when a report could be received. (If not now, it might be in the future or depending on vendor implementation?)

So if you still feel the message on the original PR was better, feel free to merge that one and close this one! I just want 'a' solution to be included ASAP.